jax/_src/dtypes.py

```diff
@@ -435,7 +435,7 @@ def _issubdtype_cached(a: type | np.dtype | ExtendedDType,
 }
 
 
-def isdtype(dtype: DTypeLike, kind: str | DType | tuple[str | DType, ...]) -> bool:
+def isdtype(dtype: DTypeLike, kind: DTypeLike | tuple[DTypeLike, ...]) -> bool:
```

Comment:
I would do `str | DTypeLike`. It's true that `DTypeLike` contains strings, but this functions allows strings that are not valid dtype identifiers.